src/Generics/SOP/Universe.hs

```diff
@@ -173,6 +191,20 @@ type IsProductType (a :: *) (xs :: [*]) =
 type IsEnumType (a :: *) =
   (Generic a, All ((~) '[]) (Code a))
 
+-- | Convert from an enum type to its sum representation.
+--
+-- @since 0.3.3.0
+--
+enumFrom :: IsEnumType a => a -> NS (K ()) (Code a)
+enumFrom = map_NS (const (K ())) . unSOP . from
```

Comment:
I to don't expect much use for these. and naming these is hard. One option is to have `Generics.SOP.Extra` module with more controversial and name-clashy stuff. Like http://hackage.haskell.org/package/lens-4.16/docs/Control-Lens-Extras.html